recipe/magento2.php

```diff
@@ -48,13 +54,14 @@
 // Tasks
 desc('Compile magento di');
 task('magento:compile', function () {
+    run('cd {{release_path}} && {{bin/composer}} dump-autoload -o');
     run("{{bin/php}} {{release_path}}/bin/magento setup:di:compile");
     run('cd {{release_path}} && {{bin/composer}} dump-autoload -o');
 });
 
 desc('Deploy assets');
 task('magento:deploy:assets', function () {
-    run("{{bin/php}} {{release_path}}/bin/magento setup:static-content:deploy {{static_content_locales}}");
+    run("{{bin/php}} {{release_path}}/bin/magento setup:static-content:deploy {{static_content_locales}} --force");
```

Comment:
Any specific reason why we're suddenly forcing this?

Reply:
Starting magento 2.2 i guess, static content deploy throws an error when env is not in production mode
I may have a reason to have my environment in developer o default mode but still want to deploy it  
--force just allows to generate the static content in any mode

Reply:
I think your case is an exception - by default, Magento should not be in developer mode (and never in default mode - that is the worst of both worlds) when deploying to production. Leaving `--force` would generate an error - exactly as it should. If you have an exception use case, you can override this in your own customizations to the recipe.

Reply:
i agree with the production envs part, but in a normal workflow we also use deployer for dev and test environments, were it's usual  to have them in developer mode in order to get more insights about errors for instance  
do you see any way to compatibilize this at least for previous environments?

Reply:
Use `on()`? https://deployer.org/docs/api.html#on

Reply:
don't see the way to identify previous envs, since there could be a wide variety on stage naming conventions: dev, development, testing, staging...  
anyway thinking twice we should not leave responsability of checking appropiate env mode to this static-content command, but maybe moving this check to an independent task when we find the way to identify non-production environments

Reply:
I'm still in favor of not putting this in the default recipe at all and leaving it up to the user of the recipe to think of a way to identify their non-production environments by overriding this task, if they want to.

Reply:
fine for me

Reply:
I would skip the --force indeed.